salt/client/ssh/__init__.py

```diff
@@ -580,7 +588,7 @@ def handle_ssh(self, mine=False):
         """
         que = multiprocessing.Queue()
         running = {}
-        target_iter = self.targets.__iter__()
+        targets_queue = list(self.targets.keys())
```

Comment:
Why is it needed to create a list here?

Reply:
It's used to put the blocked minion at the end of the list to retry it after:
https://github.com/openSUSE/salt/blob/03b0e9bd05cfeb508d78c2ed95a2710310af5cdb/salt/client/ssh/__init__.py#L632